### PR TITLE
fix(all): always call client.Close when creating a client

### LINKED ISSUE #2078: all: many samples don't call client.Close(), leading to memory leaks
The code samples for Secret Manager have "client, err := secretmanager.NewClient(ctx)" but no corresponding "defer client.Close()"  which resulted in my application, which used the code in the samples as a template, having a memory leak.

It'd be great if the sample code could make it clear in the code or comments that .Close() should be called.

Thanks!

### PR DESCRIPTION
**Pull request by tbpg**
Generated with:

```
$ grep -L Close $(grep -El 'New[^(]*Client' **/*.go) | grep -v test | xargs sed -i '/New[^(]*Client/,/}/s/}/}\ndefer client.Close()/'
$ goimports -w .
```

There were ~30 places where `defer client.Close()` was not correct (e.g. when you want to keep a client around in a Cloud Function), so I didn't update those.

I verified `goimports` and `go vet` are happy:
```
$ find . -name go.mod -execdir go vet ./... \;
```

Fixes #2078.

### CODE REVIEW

File: appengine_flexible/datastore/datastore.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	defer client.Close()
```
Comment (codyoss) [inline appengine_flexible/datastore/datastore.go:45]: Looks like this should be datastoreClient. Might be other cases like this.

File: bigtable/garbagecollection/create_family_gc_intersect.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ func createFamilyGCIntersect(w io.Writer, projectID, instanceID string, tableNam
 	if err != nil {
 		return fmt.Errorf("bigtable.NewAdminClient: %v", err)
 	}
+	defer client.Close()
```
Comment (codyoss) [inline bigtable/garbagecollection/create_family_gc_intersect.go:37]: another example of above comment.

File: functions/spanner/spanner.go [hunk 3]
```diff
@@ -47,6 +47,7 @@ func HelloSpanner(w http.ResponseWriter, r *http.Request) {
 			log.Printf("spanner.NewClient: %v", err)
 			return
 		}
+		defer client.Close()
```
Comment (codyoss) [inline functions/spanner/spanner.go:50]: This would close close too soon as is.

**Review by codyoss — CHANGES_REQUESTED** [review]
Only spot checked a couple. We will want to double check all of the serverless examples as those are written a little different than "standard" examples

File: functions/spanner/spanner.go (hunk 3 above)
Comment (tbpg) [reply functions/spanner/spanner.go]: Done. Great catch.

File: appengine_flexible/datastore/datastore.go (hunk 1 above)
Comment (tbpg) [reply appengine_flexible/datastore/datastore.go]: Done.

File: bigtable/garbagecollection/create_family_gc_intersect.go (hunk 2 above)
Comment (tbpg) [reply bigtable/garbagecollection/create_family_gc_intersect.go]: Done.

**Review by tbpg — COMMENTED** [review]
Thanks! Please take another look.

**Review by codyoss — APPROVED** [review]
Thanks, this is great!

File: logging/simplelog/simplelog.go [hunk 4]
```diff
@@ -48,11 +48,13 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed to create logging client: %v", err)
 	}
+	defer client.Close()
 
 	adminClient, err := logadmin.NewClient(ctx, projID)
 	if err != nil {
 		log.Fatalf("Failed to create logadmin client: %v", err)
 	}
+	defer client.Close()
```
[the remark this answers is not in the document]
Comment (nicoleczhu) [reply logging/simplelog/simplelog.go]: updated with a commit

**Review by nicoleczhu — APPROVED** [review]

**Comment by tbpg** [comment]
Great catch! Thank you!

**Pull request merged**